packages/amplication-design-system/src/components/PlanUpgradeConfirmation/PlanUpgradeConfirmation.tsx

```diff
@@ -13,13 +13,19 @@ export type Props = DialogProps & {
   isOpen: boolean;
   onConfirm: () => void;
   onDismiss: () => void;
+  onFirstRender: () => void;
 };
 
 export const PlanUpgradeConfirmation = ({
   isOpen,
   onConfirm,
   onDismiss,
+  onFirstRender,
 }: Props) => {
+  useEffect(() => {
+    onFirstRender();
+  }, []);
```

Comment:
why do we need the dependency array here?

Reply:
same as in useCallback, empty dependency array cause only one effect call.